### DISCUSSION TITLE
Just published version 0.7.0 of all packages and new TaskContext object for Agents

### DISCUSSION THREAD

**Discussion by Stevenic** [Stevenic/alphawave#13] [Announcements]
The update includes a new `TaskContext` interface and `AgentTaskContext` class that lets a caller manually cancel an agent mid task. It also lets commands increment the agents step count and return new thoughts to the caller. Here's an example of using the AgentTaskContext:

```
try {
    this.context = new AgentTaskContext(this.agent);
    await this.agent.completePrompt(userMessage, this.context);
} finally {
    this.context = undefined;
}
```

Then elsewhere in your code you can cancel the running task by calling `this.context?.cancel()`. The agent will stop when it finishes the current step or operation its performing. You can also inspect properties off the context object to know where the agent is in its current task.

A new `max_time` option lets you control how long time wise an agent is allowed to run. The default value is 60,000ms or 1 minute.

These changes are technically breaking changes for commands but since I'm pretty sure I'm the only one creating new commands, it shouldn't break anyone. The samples have all been updated to reflect the new `Command` interface changes.

Version 0.7.0 is on npm and also includes some other minor bug fixes.

Oh there's also two new samples, both of which use OpenAI's functions: The [09.todo-list](https://github.com/Stevenic/alphawave/tree/main/samples/09.todo-list) was contributed by @rmolinamir (thank you very much) and the [04.restaurant](https://github.com/Stevenic/alphawave/tree/main/samples/04.restaurant) sample is a port of [TypeChat example](https://github.com/microsoft/TypeChat/tree/main/examples/restaurant) to use AphaWave. The AlphaWave version improves on the TypeChat version by making it conversational (TypeChat only supports single shot translation of NL to JSON) but 80% of the code is directly from their sample. It's a great way to compare the two libraries.

**Comment by rmolinamir** [comment]
Cool stuff! Love the restaurant example. Been meaning to ask, I ran into issues inside a Lambda function environment because of my bundle, something to do with `promptrix` using `gpt-3-encoder`. Do you think it's possible to use some kind of adapter pattern to set up an encoder? For example, I'd love to use [tiktoken](https://github.com/dqbd/tiktoken), I think they have a pure JS solution.

**Reply by onhate** [reply]
@rmolinamir I had a similar issue and if you are bundling your lambda with webpack you can solve it with something like this 


```js
const CopyPlugin = require('copy-webpack-plugin');
const path = require('path');

const gpt3Path = path.dirname(require.resolve(`gpt-3-encoder/package.json`))

module.exports =  {
    ... 
    plugins: [
      new CopyPlugin({
        patterns: [
          { from: path.join(gpt3Path, 'encoder.json'), to: 'encoder.json' },
          { from: path.join(gpt3Path, 'vocab.bpe'), to: 'vocab.bpe' }
        ]
      })
    ]
};
```

**Reply by rmolinamir** [reply]
Thanks @onhate that is perfect, makes the whole bundling much easier.
